src/Rings/MPolyMap/flattenings.jl

```diff
@@ -121,6 +121,132 @@ mutable struct RingFlattening{TowerRingType<:MPolyRing, FlatRingType<:Ring,
                                                      L_to_S_flat
                                                     )
   end
+
+  function RingFlattening(
+      S::MPolyQuoRing{RingElemType}
+    ) where {RingElemType <: MPolyRingElem{<:MPolyRingElem}}
+    P = base_ring(S) # the free polynomial ring
+    P_flattening = flatten(P)
+    R = base_ring(P) # the coefficient ring of S
+    kk = coefficient_ring(R)
+
+    P_flat = codomain(P_flattening)
+    S_flat, _ = quo(P_flat, P_flattening(modulus(S)))
+
+    R_to_S_flat = hom(R, S_flat, gens(S_flat)[ngens(S)+1:end])
+    S_to_S_flat = hom(S, S_flat, R_to_S_flat, gens(S_flat)[1:ngens(S)])
+    S_flat_to_S = hom(S_flat, S, vcat(gens(S), S.(gens(R))))
+    return new{typeof(S), typeof(S_flat), typeof(R)}(S, S_flat, R, 
+                                                     S_to_S_flat, S_flat_to_S,
+                                                     R_to_S_flat
+                                                    )
+  end
+
+  function RingFlattening(
+      S::MPolyQuoRing{RingElemType}
+    ) where {RingElemType <: MPolyRingElem{<:MPolyQuoRingElem}}
+    P = base_ring(S)::MPolyRing # the polynomial ring behind S
+    A = base_ring(P)::MPolyQuoRing # the coefficient ring of S
+    R = base_ring(A)::MPolyRing # the polynomial ring behind the coefficient ring of S
+    I = modulus(A)::MPolyIdeal # the modulus of the coefficient ring of S
+    kk = coefficient_ring(R)::Field
+
+    P_flattening = flatten(P)
+    P_flat = codomain(P_flattening)
+    mod_flat = P_flattening(modulus(S))
+    S_flat, _ = quo(P_flat, mod_flat)
+
+    A_to_S_flat = hom(A, S_flat, gens(S_flat)[ngens(S)+1:end])
+
+    S_to_S_flat = hom(S, S_flat, A_to_S_flat, gens(S_flat)[1:ngens(S)])
+    S_flat_to_S = hom(S_flat, S, vcat(gens(S), S.(P.(gens(A)))))
+
+    return new{typeof(S), typeof(S_flat), typeof(A)}(S, S_flat, A, 
+                                                     S_to_S_flat, S_flat_to_S,
+                                                     A_to_S_flat
+                                                    )
+  end
+
+  function RingFlattening(
+      S::MPolyQuoRing{RingElemType}
+    ) where {RingElemType <: MPolyRingElem{<:MPolyQuoLocRingElem}}
+    P = base_ring(S)::MPolyRing
+    Q = base_ring(P)::MPolyQuoLocRing
+    R = base_ring(Q)::MPolyRing
+    L = localized_ring(Q)::MPolyLocRing
+    A = underlying_quotient(Q)::MPolyQuoRing
+    U = inverted_set(Q)::AbsMultSet
+    I = modulus(Q)::MPolyLocalizedIdeal
+    kk = coefficient_ring(R)::Field
```

Comment:
I do not really understand what happens here. Also `A`, `U` and  `I` are not used anywhere. This looks fishy. Please check.

Reply:
It would be nice to write a small comment on what the objects are. It is hard for me to understand this. 
Like ` (U^-1QQ[x])[y] / J -> U^{-1}QQ[x,y]/J`

Reply:
What happened here: I first gathered all information I thought would be useful to untangle things. Then I implemented 2-3 different versions of how to achieve it and in the end I didn't use all the data from the beginning. Cleaned it up now.